Add reqliquary strategy

0xSkly opened the pull request:
Adds reliquary strategy by

get amount of nfts per user with balanceOf
get the tokenId / relicId of those nfts
sum up weighted amount of nfts for configured pool
weighting is done with level / maxVotingLevel * amount

parameters:

reliquaryAddress - address of reliquary contract
poolId - pool id used for voting power
maxVotingLevel - level you chose for max maturity for voting
minVotingLevel - minimum level to have any voting power
multiplier - what should be used as multiplier, [level,allocationPoints]
decimals - decimals of the LP deposited in the pool

0xSkly commented:
resolves #2

franzns commented on src/strategies/reliquary/README.md:
```diff
@@ -0,0 +1,40 @@
+# reliquary
+
+This strategy utilizes Relic NFTs from Reliquary to calculate voting power. The strategy can be configured to either
+use the actual levels to weight the voting power or the allocation points assigned to each level.
```
... use the level number to weight...

franzns commented on src/strategies/reliquary/README.md:
```diff
@@ -0,0 +1,40 @@
+
+`votingPower = level / maxVotingLevel * amount`
+
+or if we use allocation ponts
```
points

franzns commented on src/strategies/reliquary/README.md:
```diff
@@ -0,0 +1,40 @@
+In other words, if the nft has reached max voting level the voting power is equal to the amount deposited.
+
+Since Relic levels only update on an interaction, we have to chose if we want to use the current 'actual' level, or the level which the relic would
+have after an update. This can be done by the `useLevelOnUpdate` flag.
```
using, not by

Commit:
Add reqliquary strategy

Use levelOnUpdate to retrieve correct level


Add improvements


rename config param to multiplier


add minimum voting level option


Adjust voting power formula


rename variable


Add readme


improve


Improve readme & rename multiplier to strategy


fix readme

Commit:
Use abi string & fix author

Commit:
Merge branch 'master' into reliquary
